pkg/storage/db.go

```diff
@@ -293,23 +268,66 @@ func (db *DB) runQuery(ctx context.Context, query string, args ...interface{}) (
 
 	defer rows.Close()
 
-	networkChangeEvents := make([]domain.NetworkChangeEvent, 0)
+	cloudAssetDetails := make([]domain.CloudAssetDetails, 0)
+
+	tempMap := make(map[string]*domain.CloudAssetDetails)
+
 	for rows.Next() {
-		var row domain.NetworkChangeEvent
-		var bytes []byte
+		var row domain.CloudAssetDetails
+
+		var metaBytes []byte
 		var hostname sql.NullString
-		err = rows.Scan(&row.ResourceID, &row.IPAddress, &hostname, &row.IsPublic, &row.IsJoin, &row.Timestamp, &row.AccountID, &row.Region, &row.Type, &bytes)
+		var ipAddress sql.NullString
+		var isPublic sql.NullBool
+		var isJoin sql.NullBool
+		var timestamp time.Time
+
+		err = rows.Scan(&row.ResourceID, &ipAddress, &hostname, &isPublic, &isJoin, &timestamp, &row.AccountID, &row.Region, &row.ResourceType, &metaBytes)
+
 		if err == nil {
-			if bytes != nil {
+			if metaBytes != nil {
 				var i map[string]string
-				_ = json.Unmarshal(bytes, &i) // we already checked for nil, and the DB column is JSONB; no need for err check here
+				_ = json.Unmarshal(metaBytes, &i) // we already checked for nil, and the DB column is JSONB; no need for err check here
 				row.Tags = i
 			}
+			if tempMap[row.ResourceID] == nil {
```

Comment:
You might want to check with @zlozano to ensure there are no collisions with his work to replace `ResourceID` with `ARN`

Reply:
I'll wait for him to merge before I start my work on this service. The merge conflicts will be :toodamnhigh: